bambi/models.py

```diff
@@ -108,49 +120,41 @@ def build(self, backend=None):
                 and look in self._backend_name.
         '''
 
+        # retain only the complete cases
+        n_total = len(self.data.index)
+        if len(self.completes):
+            completes = [set(x) for x in sum(self.completes, [])]
+            completes = set.intersection(*completes)
+        else:
```

Comment:
`len(self.completes)` should always be equal to the number of terms, because it is a list of arrays of length N, where N is the full number of rows. The if-else is just for the unlikely case of the empty model with no predictors (which is in our tests, although it's rarely of interest in the real world). If 100% of rows have a missing value, the `clean_data` DataFrame should end up empty, and the user should see a warning that N/N rows were dropped before see the eventual error.